<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <title>RNA-Seq Pipelines</title>

        <meta name="description" content="RNA-Seq pipelines and how we can build them">
        <meta name="author" content="Thasso Griebel">
        <meta name="apple-mobile-web-app-capable" content="yes" />
        <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
        <link rel="stylesheet" href="css/reveal.min.css">
        <link rel="stylesheet" href="css/theme/default.css" id="theme">
        <!-- For syntax highlighting -->
        <link rel="stylesheet" href="lib/css/zenburn.css">
        <!-- If the query includes 'print-pdf', use the PDF print sheet -->
        <script>
            document.write( '<link rel="stylesheet" href="css/print/' + ( window.location.search.match( /print-pdf/gi ) ? 'pdf' : 'paper' ) + '.css" type="text/css" media="print">' );
        </script>
        <!--[if lt IE 9]>
        <script src="lib/js/html5shiv.js"></script>
        <![endif]-->
    </head>
    <body>
        <div class="reveal">
            <div class="slides">
                <section>
                    <h1>RNA-Seq Pipelines</h1>
                    <h3>Thasso Griebel</h2>
                </section>
                <section>
                    <h2>Applications</h2>
                    <ul>
                        <li>Quantification</li>
                        <li>Assembly</li>
                        <li>Fusion Detection</li>
                        <li>Gene Model prediction</li>
                        <li>...</li>
                    </ul>
                </section>
                <section>
                    <h2>General Steps</h2>
                    <ul>
                        <li>Quality Control of the reads</li>
                        <li>Mapping</li>
                        <li>Quality Control of the mapping</li>
                        <li>Filter</li>
                        <li>...</li>
                    </ul>
                </section>
                <section>
                    <h2>FastQC</h2>
                    <h3><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></h3>
                    <img src="img/fastqc.png"/>
                </section>
                <section>
                    <h2>GEM</h2>
                    <h3><a href="http://algorithms.cnag.cat/">http://algorithms.cnag.cat</a></h3>
                </section>
                <section>
                    <h2>GEMTools</h2>
                    <h3><a href="http://gemtool.github.io">http://gemtool.github.io</a></h3>
                    <ul>
                        <li>Wrap GEM</li>
                        <li>Stats</li>
                        <li>Filter</li>
                        <li>Count</li>
                        <li>...</li>
                    </ul>
                </section>
                <section>
                    <h2>GEMTools RNA-Seq pipeline</h2>
                    <img src="img/gemtool_rnaseq_pipeline.png">
                </section>
                <section>
                    <h2>Mapping Steps</h2>
                    <p>All reads against:</p>
                    <ul>
                        <li>genome</li>
                        <li>annotation transcriptome</li>
                        <li>de-novo transcriptome</li>
                    </ul>
                </section>
                <section>
                    <h2>Filtering</h2>
                    <p>This depends on the application!</p>
                    <p>The default filter:</p>
                    <ul>
                        <li>minimum exon overlap: 5</li>
                        <li>minimum intron length: 20</li>
                        <li>Keep unique mappings</li>
                        <li>Not more than 5 alignments</li>
                    </ul>
                </section>
                <section>
                    <h2>Counts</h2>
                    <p>GEMTools can create gene-level counts</p>
                    <ul>
                        <li>given annotation</li>
                        <li>given gene model</li>
                        <li>weighted multi-maps</li>
                        <li>full exon overlap</li>
                    </ul>
                </section>
                <section>
                    <h2>Gene counts example</h2>
                    <img src="img/counts_1.png"/>
                </section>
                <section>
                    <h2>NO Multi-maps</h2>
                    <img src="img/counts_2.png"/>
                    <p>A: 6 B: 6 C: 2</p>
                </section>
                <section>
                    <h2>Multi-maps</h2>
                    <img src="img/counts_4.png"/>
                    <p>A: 10 B: 8 C: 4</p>
                </section>
                <section>
                    <h2>Weighted with Exon overlap</h2>
                    <img src="img/counts_3.png"/>
                    <p>A: 5.0 B: 3.0 C: 0</p>
                </section>
                <section>
                    <h2>Mapping QC</h2>
                    <ul>
                        <li>mapped reads</li>
                        <li>multi maps</li>
                        <li>errors</li>
                        <li>split maps</li>
                        <li>...</li>
                    </ul>
                </section>
                <section>
                    <h2>Mapped Reads</h2>
                    <img src="img/mapped_reads_abs.png"/>
                </section>
                <section>
                    <h2>Error Positions</h2>
                    <img src="img/error_pos.png"/>
                </section>
                <section>
                    <h2>Junction Positions</h2>
                    <img src="img/junction_pos.png"/>
                </section>
                <section>
                    <h2>GEMTools RNA-Seq pipeline</h2>
                    <img src="img/gemtool_rnaseq_pipeline.png">
                </section>
                <section>
                    <h2>What is a "Pipeline"</h2>
                    <p><blockquote
                        cite="http://en.wikipedia.org/wiki/Pipeline_(computing)">
                        &ldquo;Computer-related
                        pipelines include: Software pipelines, where commands
                        can be written so that the output of one operation is
                        automatically used as the input to the next, following
                        operation.&rdquo;</blockquote></p>
                </section>
                <section>
                    <h2>Example: Counts</h2>
                    <p><ul>
                        <li class="fragment">Open my compressed <i>table.gz</i> file</li>
                        <li class="fragment">Extract the second column</li>
                        <li class="fragment">Find all occurrences of "105"...</li>
                        <li class="fragment">...and count them</li>
                    </ul>
                    <pre  class="fragment" style="font-size: 0.8em;" ><code data-trim contenteditable>
                        zcat table.gz | awk '$2 == 105' |  wc -l
                    </code></pre>
                    </p>
                </section>
                <section>
                    <h2>A Pipeline</h2>
                    <img src="img/pipe_1.png"/>
                </section>
                <section>
                    <h2>Tool</h2>
                    <ul>
                        <li>smallest executable unit</li>
                        <li>defined input</li>
                        <li>defined output</li>
                    </ul>
                </section>
                <section>
                    <h2>Be precise, the small things count</h2>
                    <p class="fragment">Input file </p>
                    <code class="fragment" data-trim>
                    read_id \t count_a \t count_b
                    </code>
                    <p class="fragment">The script</p>
                    <pre class="fragment" style="font-size: 0.8em;"><code data-trim>
                    awk '{x =+ $3}END{print x}' data.txt
                    </code></pre>

                    <div class="fragment">
                    <div class="fragment" style="float:left; width:50%">
                    
                        <p>Casava < 1.8</p>
                        <pre style="font-size: 0.8em;"><code data-trim style="font-size: 0.8em;">
a/1 5 17
a/2 4 21
                        </code></pre>
                    </div>
                    <div class="fragment" style="float:left; width:50%;">
                        <p>Casava >= 1.8</p>
                        <pre style="font-size: 0.8em;"><code data-trim>
a 1:N:18  5 17
a 2:N:18  4 21
                        </code></pre>
                    </div>
                    </div>
                    <pre class="fragment" style="clear: both; font-size: 0.8em;"><code data-trim>
                    awk -F'\t' '{x =+ $3}END{print x}' data.txt
                    </code></pre>
                </section>
                <section>
                    <h2>Dependencies</h2>
                    <ul>
                        <li>Build a tree of tool (or a graph)</li>
                        <li>Execute in the proper order</li>
                        <li>Delegate data</li>
                    </ul>
                </section>
                <section>
                    <h2>Cluster integration</h2>
                    <ul>
                        <li>Mange runtimes</li>
                        <li>Threads</li>
                        <li>Memory</li>
                    </ul>
                </section>
                <section>
                    <h2>And why do we need all this</h2>
                    <ul>
                        <li>RNA-Seq experiments are large</li>
                        <li>There is no simple solution</li>
                        <li>Experiments evolve</li>
                        <li>Reproducibility</li>
                    </ul>
                </section>
                <section>
                    <h1>Thank you</h1>
                </section>
            </div>
        </div>
        <script src="lib/js/head.min.js"></script>
        <script src="js/reveal.min.js"></script>
        <script>
            // Full list of configuration options available here:
            // https://github.com/hakimel/reveal.js#configuration
            Reveal.initialize({
                controls: true,
                progress: true,
                history: true,
                center: true,
                theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
                transition: Reveal.getQueryHash().transition || 'default', // default/cube/page/concave/zoom/linear/fade/none

                // Optional libraries used to extend on reveal.js
                dependencies: [
                    { src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
                    { src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
                    { src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
                    { src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                    { src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
                    { src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
                ]
            });

        </script>

    </body>
</html>
